
syntax = "proto3";

package dunedaq.ers;

message Context {

  string cwd = 1;
  string file_name = 2;
  string function_name = 3;
  string host_name = 4;
  uint32 line_number = 5;
  string package_name = 6;

  uint32 process_id = 11;
  uint32 thread_id = 12;
  uint32 user_id = 13;
  string user_name = 14;
  string application_name = 15;
}

enum Severity {

  DEBUG = 0;
  LOG = 1;
  INFORMATION = 2;
  WARNING = 3;
  ERROR = 4;
  FATAL = 5;  
}


import "google/protobuf/timestamp.proto";

message IssueObject {

  Context context = 1;

  string name = 2;
  string message = 3;
  Severity severity = 4;

  map<string, string> parameters = 5;
  google.protobuf.Timestamp time = 6;
}

message IssueChain {

  IssueObject message = 1;
  repeated IssueObject causes = 2;
  
}